static-checks/ansible/allowed-modules/test.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python3
+
+import os
+import re
+import subprocess
+import tempfile
+import yaml
+import contextlib
+from pathlib import Path
+
+from lib import util, results
+
+
+# Obtained from
+# https://docs.ansible.com/ansible/latest/reference_appendices/playbooks_keywords.html#task
+ansible_reserved_keywords = ['action', 'any_errors_fatal', 'args', 'async', 'become', 'become_exe',
+                             'become_flags', 'become_method', 'become_user', 'changed_when',
+                             'check_mode', 'collections', 'connection', 'debugger', 'delay',
+                             'delegate_facts', 'delegate_to', 'diff', 'environment', 'failed_when',
+                             'ignore_errors', 'ignore_unreachable', 'local_action', 'loop',
+                             'loop_control', 'module_defaults', 'name', 'no_log', 'notify', 'poll',
+                             'port', 'register', 'remote_user', 'retries', 'run_once', 'tags',
+                             'throttle', 'timeout', 'until', 'vars', 'when', 'with_items']
+
+
+@contextlib.contextmanager
+def select_all_rules(datastream):
+    """
+    Given an XML file path as 'datastream' enable all the rules in the datastream.
+
+    Use as a context manager, returning a filesystem path to the
+    modified datastream.
+    """
+    exprs = set()
+    exprs.add(re.compile(r'<.*Rule.*id="'))
+    exprs.add(re.compile(r'<.*select.*idref="'))
+
+    with open(datastream) as ds:
+        with tempfile.NamedTemporaryFile(mode='w', suffix='.xml') as new_ds:
+            for line in ds:
+                if any(x.search(line) for x in exprs):
+                    line = line.replace('selected="false"', 'selected="true"')
+                new_ds.write(line)
+            new_ds.flush()
+            util.log(f"Saving modified {datastream} with all rules being selected as"
+                     "{new_ds.name}")
+            yield Path(new_ds.name)
+
+
+def list_append_ansible_module_from_task(modules_list, task_keywords, allowed_modules):
```

Comment:
Are you sure you want this as a separate function?

It's called only from `traverse_blocks()` (where it's contained in one `else:` block), it gets all its arguments, etc. .. It could be better to just include the function body in `traverse_blocks()`.